client/src/reducers/datasets.js

```diff
@@ -1,6 +1,7 @@
 import * as constants from '../constants/dataset';
 
 export const initialState = {};
+export const FETCHING = 'FETCHING';
```

Comment:
I know its a bit weird to use a string in place of a dataset object but I couldnt think of a better place to store this data (dataset is being fetched) in the state semantically-speaking. @gabemart Im open to suggestions

Reply:
I agree it feels a bit off. I'm also seeing some errors in the visualisation error when switching datasets, I think because there is some filtering code that is still expecting an object and chocking when it gets a string. I also think we should avoid filtering datasets out of the library view even when they are mid-fetch.

I can think of two alternate approaches. The first is to simply not store the `FETCHING` status at all, and infer it wherever we need it (i.e. something like `Boolean(dataset.get('rows'))` would tell us if the dataset is loaded or not. This has the advantage of not storing additional state, but it has the disadvantage that we can no longer infer if we *need* the dataset in question (whereas storing an explicit `FETCHING` state at the point of requesting a dataset also implicitly records that something requires the dataset to be loaded).

The second approach, which I think I prefer, is to move the `FETCHING` state to the existing `status` property of the dataset object. Something like

````
function fetchingDataset(state, action) {
  const { id } = action;

  return {
    ...state,
    [id]: state[id] ? state[id].set('status', FETCHING) : null,
  };
}
````

This would allow us to revert the changes to `LibraryListing.jsx`. 

It feels a bit impure, because `FETCHING` isn't really a property of the dataset but rather a property of the client - but I think on balance it has the fewest disadvantages. The changes to the containers to check the status should hopefully be fairly minor.